lib/devise/token_authenticatable/model.rb

```diff
@@ -32,7 +32,12 @@ module TokenAuthenticatable
       module ClassMethods
 
         def find_for_token_authentication(conditions)
-          find_for_authentication(authentication_token: conditions[Devise::TokenAuthenticatable.token_authentication_key])
+          conditions = conditions.dup
```

Comment:
Please, don't overwrite a method argument. Just name the variable differently, e.g., `auth_conditions`.